### DISCUSSION TITLE
How to collaborate with ASReview?

### DISCUSSION THREAD

**Discussion by Rensvandeschoot** [asreview/asreview#975] [Show and tell]
Systematic reviewing using software implementing Active Learning (AL) is relatively new. Many users (and reviewers) have to get familiar with the many different ways of how AL can be used in practice. In this discussion thread, we will discuss various options meant to inspire others and help answer questions from your collaborators or reviewers. 

Let’s assume you conducted a systematic search in multiple databases, the results were merged into one dataset, the data was de-duplicated, and as many abstracts as possible were retrieved ([why?](https://asreview.nl/blog/the-importance-of-abstracts/)). You found 10,000 potentially relevant records, and you want to screen the records based on predefined inclusion/exclusion criteria with two screeners, researchers A and B. Also, you have ten records of which you already know are relevant and ten irrelevant; this is what we call _prior knowledge_. 

What options do you have?

**Comment by Rensvandeschoot** [comment]
Both researchers install the software locally, upload the same data and precisely the same prior knowledge (e.g., ten relevant +10 irrelevant), train the same model, and independently start screening for relevance. After both researchers are done screening (either because the time is up or because the model presents only irrelevant records), the results, containing the labeling decisions and the ranking of the unseen records, are exported as a RIS, CSV, or XLSX file. Both files can be merged in, for example, Excel or R. Now, you can compute the similarity in decisions, just like with a classical PRISMA-based review. The only difference is that there might be records only seen by one of the two researchers. My advice would be to discuss such papers with the team and make a joint decision.

**Reply by Rensvandeschoot** [reply]
One researcher can set up the project and share the project before the actual screening starts. This will make sure that you really start with the same priors and project setup.

**Comment by Rensvandeschoot** [comment]
Both researchers use the same data but different settings: for example different prior knowledge so that you can check if using different starting values result in the same set of relevant papers.  A similar procedure can be applied by choosing different models (there are many different feature extraction techniques and classifiers available).

**Comment by Rensvandeschoot** [comment]
Sequential Screening:

Research A starts with screening and Reviewer B takes over if the screening time for A is done. If B is done, then A takes over again. (this is especially interesting if you collaborate with researchers in different time zones, so screening can continue 24 hours per day!). You can export the project file, share it with a colleague who can import the file in ASReview to continue screening where the first researchers stopped (officially supported). 

Alternatively, ASReview can be put on a server (not officially supported, but successfully implemented by some users).

**Comment by Rensvandeschoot** [comment]
Researcher A starts with screening until only irrelevant records are presented. Researcher B imports the same dataset but uses different prior knowledge: all relevant records found by Researcher A as well as the unseen records, but not the excluded records. This can easily be done in Excel (or R) by adding a column to the dataset containing the labeling decisions which should be used as prior knowledge. Research B trains a new model, and screens until only irrelevant records are found. This strategy is a quality check for the labeling decisions made by the first screener (i.e., to check for incorrectly excluded records).

**Comment by Rensvandeschoot** [comment]
After researcher A is done, the data is exported and imported again (the software will automatically recognize the previous labeling decisions and will use these as prior knowledge to train the first iteration of the model). Researcher B trains a different model, preferably with a more advanced feature extractor (e.g., sBert, doc2vec) in combination with a neural network (e.g., 2NN, LSTM, 17-layer CNN). Such deep learning models typically dig deeper into the contextual meaning of the text but require much more training data. Researcher B trains a deep learning model (this might take some time, up to a couple of days depending on the size of your data and your pc (you might want to run this model on a [cloud] server), and screens another set of records.

**Comment by langtonhugh** [comment]
Once researchers A and B have completed their reviews, they might be interested in this: an R Markdown file that automatically generates a report summarising the level of agreement between their decisions. You can view all the material required to generate your own report on the corresponding [GitHub repository](https://github.com/langtonhugh/asreview_irr) and look at a demonstration report on the [html preview page](https://htmlpreview.github.io/?https://github.com/langtonhugh/asreview_irr/blob/main/report_example.html).

A number of descriptives are produced in the report, including the number of abstracts reviewed, the number flagged as relevant, the number left unreviewed, the Kappa statistic, and the number of abstracts for which researchers A and B disagreed (e.g., A said relevant, B said irrelevant). In the final case, tables are produced which detail the specific literature involved.

The report also identifies the number of cases in which there was disagreement 'implicitly'. This is when one researcher rates an abstract as relevant, but the other researcher never reviews that abstract. This is likely to occur because of the 'stop rule' used by the team (e.g., one hundred irrelevant articles in a row). If this number is greater than zero for either researcher, it indicates that one or both researchers stopped too early.

The report is a work in progress. I would appreciate any checks, feedback, comparisons to your own findings, and suggestions for other features the report could include.

**Reply by robintschoetschel** [reply]
That's a helpful resource! But don't you think that providing Kappa/IRR scores based on only the actually seen/reviewed records would be informative as well? Using the full dataset inflates the percentage agreement for sure and might bias Kappa as well (I know it corrects against chance agreement). I think metrics of both have their use: the actual rater agreement tells us more about how much human raters actually agree on the inclusion criteria, and the scores you calculate give an indication of how well that information is used by the algorithm. That's at least how I interpret it.

**Reply by Rensvandeschoot** [reply]
Do you mean basing the IRR on the set of records seen by both screeners?

**Reply by robintschoetschel** [reply]
Hi Rens, yes, exactly, I think that would make sense as an additional piece of information. If two screeners have strongly differing views on what to include/exclude, I would be worried that the separately trained models could lead to a number of articles not seen by either screener where there is false agreement that these should be excluded. But maybe this reasoning is unfounded. This could be tested in a simulation study?

**Reply by Rensvandeschoot** [reply]
That's an interesting perspective! You've raised a valid point about the potential value of evaluating the Inter-Rater Reliability (IRR) based on the set of records screened by both reviewers. This could indeed provide additional insights into the agreement regarding the inclusion criteria between the reviewers, as you've rightly noted.

However, when considering missing data, the dynamics might change, and I'm not currently aware of any study that has delved into exploring the IRR in this specific context. Your suggestion about a simulation study seems like a promising way to shed light on this issue and I appreciate the thoughtfulness behind it.

There are some notable experts in the field of missing data that I am acquainted with. Given the unique nature of this issue, I believe their insights could significantly contribute to this discussion. I'll reach out to them and see what they think.

**Reply by Hanedinges** [reply]
@langtonhugh, I'm interested in the R Markdown file you mention .... however I never know how to continue after clicking on any GitHub link: I start with eagerness to find cool stuff and close the tab with my tail between the legs. I have never learnt how to pursue and don't understand the instructions there. I feel a painful distance between me and the 'smart techies'. However, I don't want to give up, I love everything that adds to clearity and efficiency....would you mind to add a little bit more instruction for people like me? Or maybe you can help me by sending me an existing step by step guide?  My apologizes if I have overlooked what I'm asking for...

**Comment by boer0107-zz** [comment]
For inspiration: we have a [list of systematic reviews](https://www.zotero.org/groups/4597652/asreview_public/collections/IWU8ATMK/tags/multiple_screeners/collection) with multiple screeners where ASReview was used as a screening tool.

**Comment by JasikaP** [comment]
How can I share my workinglist with another screener so we can include and exclude together?

**Reply by Rensvandeschoot** [reply]
Do you want both to label the same data? Note that this is different than the PRISMA guidelines where two screeners screen independent from each other

**Reply by JasikaP** [reply]
yes we want to label the same data together.

**Reply by Rensvandeschoot** [reply]
also, at the same time or serial?

**Reply by JasikaP** [reply]
At the same time, would that be possible?

**Comment by robintschoetschel** [comment]
Thanks for doing that — I am curious what they have to say. It might not be a big issue, as the two differently trained models might accurately reveal the disagreements between screeners in the pool. But maybe there is some bias. I am not an expert in this either, but a concerned user, so if you can post something based on what your contacts say, I’d appreciate it.

**Comment by RBourry** [comment]
I have a question regarding how ASReview lab applies stopping rules when screening with multiple reviewers. 

Background:
I'm currently planning a systematic review for meta-analysis in the context of stroke care. We will screen with 2-4 reviewers and our preliminary search from just PubMed contains about 9,000 results, which will be augmented by more databases in the upcoming weeks. To ensure a recall which is as high as possible while also keeping screening time feasible we developed a workflow with stopping heuristics which are strongly oriented towards the SAFE-Framework while also trying to implement input from the recent stellar tutorial by Fütterer et al.
To ensure the training dataset is as good as it possibly can be, we'll first manually screen a randomly sampled subset of 200 records, while ensuring that random sampling will be stratified by publication date. We'll also use that phase to maximize inter-rater-agreement and then use this harmonized result of phase 1 as prior knowledge in phase 2.

Now my question:
In phases 2-4, if I understood correctly, the active-learning / neural-network models will give each reviewer a separate subset of the remaining unlabeled records. So, when defining the cutoffs for consecutive records labeled as irrelevant (e.g. 100), are those intended to be applied on an individual reviewer level or on the combined review level? So if we predefine 100 consecutive irrelevant records with 4 reviewers, does the stopping rule trigger in ASReview lab when all reviewers combined reach 100 (e.g. 1: 30, 2: 20, 3: 40, 4: 10), or when every single reviewer has reached 100 consecutive irrelevant records?

Sorry for the long post and thank you so much for your time and consideration!

TL;DR: When screening with multiple reviewers, is stopping rule "100 consecutive irrelevant records" intended to be applied per reviewer or combined across all reviewers?

**Reply by michi-zuri** [reply]
Hi @RBourry I think I can answer this for you: 

* When starting phase 2, each reviewer gets assigned the first top-ranked article that has not been assigned to anyone else yet. 
* There are not subsets assigned to specific users.
* Each article gets assigned to only one reviewer. 
* The stopping rule applies to all reviewers combined. 
* But not necessarily the way you described. If Alice and Bob collaborate and Alice has exlucded the last 80 articles, while Bob was on a break and Bob then starts screening again and includes even just 1 article, then the counter resets for everyone. So if Alice is on a break while Bob continues the screening and his first decision is an inclusion, Bob would need 100 excluded articles in a row, not 20. 
* The way you described the stopping rule, it could be interpreted that if Alice's 80 last decisions were exclusions and then stops, any other reviewer would only need 20 more exclusions in a row for the stopping rule to apply. This is not how the stopping rule is implemented at the moment, as far I can tell. What matters is the sequence of the last decisions chronologically, by all reviewers combined. 
* The downside of this could be, having to exclude 100 articles again even though you'd be finished if everyone worked synchronously: one of your 4 reviewers (let's call her Alice) may have only been active on the first day and then checks in again when everyone thinks they are done (stopping rule was nearly triggered), but then her last assigned article turns out to be relevant (after all that article was highly ranked and assigned for review on day 1, so that's quite a likely scenario). 

Hope this helps.

**Reply by RBourry** [reply]
Hey @michi-zuri thank you for the explanation!
Actually, what you're describing is exactly what I meant by the stopping rule applying across "all reviewers combined", thank you for clarifying!
The only thing I'm slightly unsure if I understood it correctly is the case you described in your last paragraph. Is the reason, why this  can occur, that if "Alice" screens e.g. 30 records and then pauses, their 31st record (which will likely correctly be ranked very high by the model) is already assigned to them immediately after they take the screening decision on record #30, and will stay assigned to them and not get re-assigned to another reviewer, even if other reviewers continue screening while Alice is not logged in?

Thank you again for the support!

**Reply by michi-zuri** [reply]
Yes that's exactly what I meant! You're welcome.

**Comment by martinewilbers** [comment]
Hi,

I am conducting a scoping review and would like to use ASReview for the screening process (with approximately 11.000 records). We would like to apply the SAFE procedure. My plan is to involve a second reviewer in phase 1 and part of phase 2 for calibration purposes. I would then complete the remainder of Phase 2 and Phase 3 myself, while the second reviewer would be involved again in Phase 4.
Would this be an appropriate way to implement the SAFE procedure?

Kind regards,
Martine
